### PR TITLE
Fixed issue related to publish the package

### PR DESCRIPTION
**Pull request by agamkrbit**
Below changes are made int the code
- Added publishing related configuration in `pom.xml` file
- Fixed error handling of code

### CODE REVIEW

File: src/main/java/com/eppo/sdk/helpers/Shard.java [hunk 1]
```diff
@@ -16,9 +16,9 @@ public class Shard {
      *
      * @param input
      * @return
-     * @throws Exception
+     * @throws RuntimeException
      */
-    public static String getHex(String input) {
+    public static String getHex(String input) throws RuntimeException {
```
Comment (ploomis14) [inline src/main/java/com/eppo/sdk/helpers/Shard.java:21]: `RuntimeException` is not a checked exception, so I don't think it should be part of the method signature. Same with the other places it is added in EppoClient.java.

File: pom.xml [hunk 2]
```diff
@@ -4,7 +4,7 @@
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
-    <groupId>com.eppo.sdk</groupId>
+    <groupId>cloud.eppo</groupId>
     <artifactId>eppo-server-sdk</artifactId>
     <version>1.0.0</version>
```
Comment (ploomis14) [inline pom.xml:9]: Please also address [this TODO](https://github.com/Eppo-exp/java-server-sdk/blob/main/src/main/java/com/eppo/sdk/EppoClient.java#L189) about the version

File: src/main/java/com/eppo/sdk/helpers/Shard.java (hunk 1 above)
Comment (agamkrbit) [reply src/main/java/com/eppo/sdk/helpers/Shard.java]: Thanks, removed the `throws RuntimeException`

File: src/main/java/com/eppo/sdk/EppoClient.java [hunk 3]
```diff
@@ -38,13 +38,13 @@ private EppoClient(ConfigurationStore configurationStore, Timer poller, EppoClie
...
      * @return
-     * @throws Exception
+     * @throws RuntimeException
      */
     public Optional<String> getAssignment(
             String subjectKey,
             String experimentKey,
             SubjectAttributes subjectAttributes
-    ) {
+    ) throws RuntimeException {
```
Comment (ploomis14) [inline src/main/java/com/eppo/sdk/EppoClient.java:47]: Remove RuntimeException from the signature here too

File: src/main/java/com/eppo/sdk/EppoClient.java [hunk 4]
```diff
@@ -152,7 +152,7 @@ private Variation getAssignedVariation(
     private String getSubjectVariationOverride(
             String subjectKey,
             ExperimentConfiguration experimentConfiguration
-    ) {
+    ) throws RuntimeException {
```
Comment (ploomis14) [inline src/main/java/com/eppo/sdk/EppoClient.java:155]: Also remove RuntimeException from the signature here

File: src/main/java/com/eppo/sdk/helpers/AppDetails.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
+package com.eppo.sdk.helpers;
+
+import java.io.*;
```
Comment (ploomis14) [inline src/main/java/com/eppo/sdk/helpers/AppDetails.java:3]: Don't use wildcard imports

File: src/main/java/com/eppo/sdk/EppoClient.java [hunk 6]
```diff
@@ -95,7 +95,7 @@ public Optional<String> getAssignment(
      * @return
      * @throws Exception
      */
-    public Optional<String> getAssignment(String subjectKey, String experimentKey) {
+    public Optional<String> getAssignment(String subjectKey, String experimentKey) throws Exception {
```
Comment (ploomis14) [inline src/main/java/com/eppo/sdk/EppoClient.java:98]: I don't think we should have a checked exception as part of this method signature; it is more difficult for consumer, as they have to use a try/catch block to call our API.

If the method body can throw an exception, it should be RuntimeException or something that inherits from RuntimeException.

File: src/main/java/com/eppo/sdk/helpers/AppDetails.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+import java.util.*;
+
+public class AppDetails {
+
+    static AppDetails instance;
+    private String version;
+    private String name;
+
+    public static AppDetails getInstance () {
```
Comment (ploomis14) [inline src/main/java/com/eppo/sdk/helpers/AppDetails.java:12]: I don't see where this is used - are you going to use it [here](https://github.com/Eppo-exp/java-server-sdk/blob/main/src/main/java/com/eppo/sdk/EppoClient.java#L189)?

File: src/main/java/com/eppo/sdk/helpers/AppDetails.java (hunk 5 above)
Comment (agamkrbit) [reply src/main/java/com/eppo/sdk/helpers/AppDetails.java]: Fixed it!

File: src/main/java/com/eppo/sdk/helpers/AppDetails.java (hunk 7 above)
Comment (agamkrbit) [reply src/main/java/com/eppo/sdk/helpers/AppDetails.java:13]: I missed pushing some code. We are using it in `EppoClient` class to get name and version.
Pushed the whole code

File: src/main/java/com/eppo/sdk/EppoClient.java (hunk 6 above)
Comment (agamkrbit) [reply src/main/java/com/eppo/sdk/EppoClient.java]: Yeah make sense! Removed the throw Exception part

**Review by ploomis14 — DISMISSED** [review]

**Review by ploomis14 — COMMENTED** [review]
Please fix failing tests:

```
Illegal character in query at index 96: http://localhost:4001/randomized_assignment/config?apiKey=mock-api-key&sdkVersion=1.0.1&sdkName="java-server-sdk"&
```

**Comment by agamkrbit** [comment]
> Please fix failing tests:
> 
> ```
> Illegal character in query at index 96: http://localhost:4001/randomized_assignment/config?apiKey=mock-api-key&sdkVersion=1.0.1&sdkName="java-server-sdk"&
> ```

Done

**Pull request merged**